app/src/main/java/at/roteskreuz/stopcorona/model/repositories/InfectionMessengerRepository.kt

```diff
@@ -303,13 +340,19 @@ class InfectionMessengerRepositoryImpl(
 
     private suspend fun fetchDailyBatchesDiagnosisKeys(dailyBatches: List<ApiDiagnosisKeysBatch>)
         : List<DbDailyBatchPart> {
-        return dailyBatches.flatMap { dailyBatch ->
-            dailyBatch.batchFilePaths.mapIndexed { index, path ->
-                DbDailyBatchPart(
-                    batchNumber = index,
-                    intervalStart = dailyBatch.intervalToEpochSeconds,
-                    fileName = apiInteractor.downloadContentDeliveryFile(path)
-                )
+        return coroutineScope {
+            dailyBatches.flatMap { dailyBatch ->
+                dailyBatch.batchFilePaths.mapIndexed { index, path ->
+                    async {
+                        DbDailyBatchPart(
+                            batchNumber = index,
+                            intervalStart = dailyBatch.intervalToEpochSeconds,
+                            fileName = apiInteractor.downloadContentDeliveryFile(path)
+                        )
+                    }
+                }.map {
+                    it.await()
```

Comment:
Do we want to continue with remaining async tasks when some of them failed? If so, please call `awaitAll()`.

Reply:
No, we don't want that. If any of the downloads fail we consider the batch failed.